packages/api/src/platforms/vtex/resolvers/searchResult.ts

```diff
@@ -45,7 +46,9 @@ export const StoreSearchResult: Record<string, Resolver<Root>> = {
 
     const skus = productSearchResult.products
       .map((product) => {
-        const [maybeSku] = product.items
+        const maybeSku = product.items.find((item) =>
+          item.sellers.some((item) => inStock(item.commertialOffer))
+        )
```

Comment:
I just realized one thing: if a product has no available SKUs but matches the search query, previously, it would return as an Out Of Stock item.

The proposed logic would make it not appear at the search page in those cases.

I think for us to match the same behavior and fix the issue, we should make something like the suggestion below. I think we have to change this in v1 as well.

```suggestion
        const maybeSku = product.items.find((item) =>
          item.sellers.some((item) => inStock(item.commertialOffer))
        ) ?? product.items[0]
```

Reply:
Good catch. we kind of do that in the [product query](https://github.com/vtex/faststore/blob/5479bd5bbee9b784a96787a55006f6b8449a948c/packages/api/src/platforms/vtex/utils/sku.ts#L22), but it is missing for `products` and `suggestions`.

Done [here](https://github.com/vtex/faststore/pull/2141/commits/29640779005f3ac2e197960baf9e011b91771314)